internal/installercache/installercache_test.go

```diff
@@ -118,20 +144,161 @@ var _ = Describe("installer cache", func() {
 			Expect(l.extractDuration).ShouldNot(BeZero())
 		}
 		Expect(l.Path).ShouldNot(BeEmpty())
-
-		time.Sleep(1 * time.Second)
 		Expect(l.startTime.Before(time.Now())).To(BeTrue())
+		l.Cleanup(context.TODO())
 		return fname, l.Path
 	}
-	It("evicts the oldest file", func() {
+
+	type launchParams struct {
+		releaseID string
+		version   string
+		clusterID strfmt.UUID
+	}
+
+	// runParallelTest launches a batch of fetches from the installer cache in order to simulate multiple requests to the same node
+	// releases are automatically cleaned up as they are gathered
+	// returns the first error encountered or nil if no error encountered.
+	runParallelTest := func(params []launchParams) error {
+		var wg sync.WaitGroup
+		var firstError error
+		var errMutex sync.Mutex
+		for _, param := range params {
+			mockReleaseCalls(param.releaseID, param.version)
+			wg.Add(1)
+			go func(param launchParams) {
+				defer wg.Done()
+				release, err := manager.Get(ctx, param.releaseID, "mirror", "pull-secret", mockRelease, param.version, param.clusterID)
+				if err != nil {
+					errMutex.Lock()
+					if firstError == nil {
+						firstError = err
+					}
+					errMutex.Unlock()
+					return
+				}
+				// Simulate calls to release cleanup to ensure that we can test capacity limits
+				// Add a very small delay so that we can simulate usage of the release prior to release.
+				if release != nil {
+					time.Sleep(2 * time.Millisecond)
+					release.Cleanup(context.TODO())
+				}
+			}(param)
+		}
+		wg.Wait()
+		return firstError
+	}
+
+	getUsedBytesForDirectory := func(directory string) uint64 {
+		var totalBytes uint64
+		seenInodes := make(map[uint64]bool)
+		err := filepath.Walk(directory, func(path string, fileInfo os.FileInfo, err error) error {
```

Comment:
Don't we already have a function that does this? Why are we duplicating it here?

Reply:
It's best practice to not use a function from production code as this may be the function that has a bug in it and to use the same function in a test would sway the test.

So repetition of code here is acceptable IMO.

Reply:
> You have a test for that function that tests that. Repetition of this code adds zero value, 
I have been given negative reviews by @carbonin in the past for using production code in this way (even when it was covered by tests.) 

>infact forces us to know that this function is here and to modify it in the case (however unlikely) that this changes.
I'd argue the same for the fragility of changing a method in production code.

But let me know what is most idiomatic, I'm confused by the variable standards on this.

Reply:
Yeah, I think my concern is more where there's some function that is actually part of what's being tested also being used to validate the output of the test.

That isn't really the case here because the function is an external (to the package) utility that is tested in isolation.

I don't care strongly about how you resolve this though. Leaving it here or using the utility makes no difference to me.

Reply:
If OK, I will keep this as is for now, this may change if I refactor the tests to work with parallel table.